llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -1656,6 +1656,11 @@ SDValue SelectionDAG::getConstant(const ConstantInt &Val, const SDLoc &DL,
   EVT EltVT = VT.getScalarType();
   const ConstantInt *Elt = &Val;
 
+  // Vector splats are explicit within the DAG, with ConstantSDNode holding the
+  // to be splatted scalar ConstantInt.
+  if (isa<VectorType>(Elt->getType()))
```

Comment:
Check VT.isVector instead? I assume these have to match?

Reply:
They don't have to match. In most cases `Val` will be the scalar you want to splat.  The new use case is when SelectionDAGBuilder passes in the new style ConstantInt which can have a vector result type when `use-constant-int-for-scalable-vector-splat` is enabled.